### PR TITLE
TST: add validation checks on levels keyword from pd.concat

### LINKED ISSUE #46653: TST: add validation checks on levels keyword from pd.concat
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
>>> df1 = pd.DataFrame({"A": [1]}, index=["x"])
>>> df2 = pd.DataFrame({"A": [1]}, index=["y"])
>>> pd.concat([df1, df2], keys=["x", "y"], levels=[["x", "y", "y"]])
     A
x x  1
y y  1


```python
>>> df1 = pd.DataFrame({"A": [1]}, index=["x"])
>>> df2 = pd.DataFrame({"A": [1]}, index=["y"])
>>> pd.concat([df1, df2], levels=[["x", "y"]])
   A
x  1
y  1
```

### Issue Description

First, we should raise when levels duplicated. Second, when levels specified and keys not given, then we should also raise since this is unreasonable. (DOC only mentions infering levels from keys but not infering keys from levels)

### Expected Behavior

```python
>>> df1 = pd.DataFrame({"A": [1]}, index=["x"])
>>> df2 = pd.DataFrame({"A": [1]}, index=["y"])
>>> pd.concat([df1, df2], keys=["x", "y"], levels=[["x", "y", "y"]])
ValueError: Level values not unique: ['x', 'y', 'y']
```

```python
>>> df1 = pd.DataFrame({"A": [1]}, index=["x"])
>>> df2 = pd.DataFrame({"A": [1]}, index=["y"])
>>> pd.concat([df1, df2], levels=[["x", "y"]])
ValueError: levels supported only when keys not None
```

### Installed Versions

<details>

1.4.1

</details>

### PR DESCRIPTION
**Pull request by GYHHAHA**
- [x] closes #46653
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v1.5.0.rst [hunk 1]
```diff
@@ -92,6 +92,9 @@ Other enhancements
 - :class:`Series` and :class:`DataFrame` with ``IntegerDtype`` now supports bitwise operations (:issue:`34463`)
 - Add ``milliseconds`` field support for :class:`~pandas.DateOffset` (:issue:`43371`)
 - :meth:`DataFrame.reset_index` now accepts a ``names`` argument which renames the index names (:issue:`6878`)
+- :meth:`pd.concat` now raises when levels is given but keys is None (:issue:`46653`)
```
Comment (mzeitlin11) [inline doc/source/whatsnew/v1.5.0.rst:95]:
```suggestion
- :meth:`pd.concat` now raises when ``levels`` is given but ``keys`` is None (:issue:`46653`)
```

File: doc/source/whatsnew/v1.5.0.rst [hunk 2]
```diff
@@ -92,6 +92,9 @@ Other enhancements
...
+- :meth:`pd.concat` now raises when levels duplicated (:issue:`46653`)
```
Comment (mzeitlin11) [inline doc/source/whatsnew/v1.5.0.rst:96]:
```suggestion
- :meth:`pd.concat` now raises when ``levels`` contains duplicate values (:issue:`46653`)
```

File: pandas/core/reshape/concat.py [hunk 3]
```diff
@@ -668,6 +668,8 @@ def _get_concat_axis(self) -> Index:
             return idx
 
         if self.keys is None:
+            if self.levels is not None:
+                raise ValueError("levels supported only when keys not None")
```
Comment (mzeitlin11) [inline pandas/core/reshape/concat.py:672]:
```suggestion
                raise ValueError("levels supported only when keys is not None")
```

File: pandas/tests/reshape/concat/test_index.py [hunk 4]
```diff
@@ -371,3 +371,24 @@ def test_concat_with_key_not_unique(self):
...
+        df2 = DataFrame({"A": [1]}, index=["y"])
+        msg = "levels supported only when keys not None"
+        with pytest.raises(ValueError, match=msg):
+            concat([df1, df2], levels=[["x", "y"]])
+
+        df1 = DataFrame({"A": [1]}, index=["x"])
+        df2 = DataFrame({"A": [1]}, index=["y"])
+        msg = "levels supported only when keys not None"
+        with pytest.raises(ValueError, match=msg):
+            concat([df1, df2], levels=[["x", "y", "y"]])
```
Comment (mzeitlin11) [inline pandas/tests/reshape/concat/test_index.py:394]: Could parameterize on levels to deduplicate a bit?

**Review by mzeitlin11 — COMMENTED** [review]
Thanks @GYHHAHA for the PR! Some small comments

**Comment by jreback** [comment]
thanks @GYHHAHA very nice!

**Pull request merged**